packages/dom-adapters/src/CaretAdapter/index.ts

```diff
@@ -0,0 +1,128 @@
+import { useSelectionChange } from '../caret/utils/useSelectionChange.js';
+import type { Caret, EditorJSModel, DataIndex, TextIndex, CaretManagerEvents, Index } from '@editorjs/model';
+import { getAbsoluteRangeOffset, getBoundaryPointByAbsoluteOffset } from '../utils/index.js';
+import type { TextRange } from '@editorjs/model';
+import { EventType } from '@editorjs/model';
+
+function hashIndex(index: Readonly<Index>): string {
+  return JSON.stringify(index);
+}
+
+function parseIndex<T extends Index>(hash: string): T {
+  return JSON.parse(hash) as T;
+}
+
+function areIndexesEqual(a: Readonly<Index | null>, b: Readonly<Index | null>): boolean {
+  if (a === null || b === null) {
+    return a === b;
+  }
+
+  return hashIndex(a) === hashIndex(b);
+}
+
+/**
+ *
+ */
+export class CaretAdapter extends EventTarget {
+  #container: HTMLElement;
+  #model: EditorJSModel;
+  #inputs = new Map<string, HTMLElement>();
+  #userCaret: Caret;
+
+  /**
+   *
+   * @param container
+   * @param model
+   */
+  constructor(container: HTMLElement, model: EditorJSModel) {
+    super();
+
+    this.#model = model;
+    this.#container = container;
+    this.#userCaret = this.#model.createCaret();
```

Comment:
probably some user identifier should be passed from somewhere or generated here. Since there can be several user carets